### ISSUE TITLE
Add certificate caching

### ISSUE THREAD

**Issue by LucasMoskun** [devinstewart/sns-payload-validator#41]
Request to add certificate caching so that it does not need to request the certificate every time it validates. See [line 82 here](https://gist.github.com/jimmoffet/ca8dd87309f878e5bd8ffb8903e08702) for an example of this in a python implementation.

**Comment by devinstewart** [comment]
Great thought @LucasMoskun.  Will most likely use [lru-cache](https://www.npmjs.com/package/lru-cache), as there is no need to reinvent the caching wheel.

**Comment by LucasMoskun** [comment]
Awesome!

**Comment by devinstewart** [comment]
Yeah, it will be.  Did some noodling in my head as I drifted off last night.  Unfortunately, this will have to change the constructor, so that the cache stays in scope.

Example: 
```javascript
const Validator = require('sns-payload-validator');
const validator = new Validator({ useCache: true, maxCerts: 1000 });

const validPayload = await validator.validate(payloadFromRequest);
```

That said, I still think this is a worthwhile feature.  Just wanted to give you the heads up that it may take through next weekend to get it implemented and all of the tests configured correctly.  Stay tuned!!

**Comment by LucasMoskun** [comment]
Right on. Let me know if there's anything I can do to help.

**Comment by devinstewart** [comment]
@LucasMoskun - Feel free to check out #43 and leave comments.  I will address any concerns you may bring up.  If we are all good, I will test in my own AWS Account tomorrow night and then ship it. :shipit: 

I live in the Central timezone in the US, to keep track of what _tomorrow night_ means.

**Comment by devinstewart** [comment]
Caching has been released to NPM in v2.0.0.  Thank you for your interest in this project.  Feel free to leave any feedback or open a new issue.

**Comment by LucasMoskun** [comment]
> Caching has been released to NPM in v2.0.0. Thank you for your interest in this project. Feel free to leave any feedback or open a new issue.

Looks great, I'll let you know if anything pops up in production. Thanks for smashing this through!

**Comment by devinstewart** [comment]
@LucasMoskun - Great to hear.  Not surprised by the TypeScript thing.  I type my modules for the VsCode intellisense, I'm not a big TypeScript person.

Unfortunately, due to NPM rules, I will have to release a v2.0.1 to fix the README, but probably worth doing.  Created issue #45.  Thanks for the catch
